src/typing.h

```diff
@@ -0,0 +1,24 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+
+namespace vsag {
+
+using InnerIdType = uint32_t;  // inner id's type; index's vector count may less than 2^31 - 1
+using LabelType = uint64_t;    // external id's type
```

Comment:
@LHT129  ID type should change to signed integer to avoid type-casting while calling from library interfaces